server/pulp/server/async/celery_instance.py

```diff
@@ -40,13 +40,21 @@
     },
 }
 
-
 celery.conf.update(CELERYBEAT_SCHEDULE=CELERYBEAT_SCHEDULE)
 celery.conf.update(CELERYBEAT_SCHEDULER='pulp.server.async.scheduler.Scheduler')
 celery.conf.update(CELERY_WORKER_DIRECT=True)
 celery.conf.update(CELERY_TASK_SERIALIZER='json')
 celery.conf.update(CELERY_ACCEPT_CONTENT=['json'])
 
+"""
+Configures the celery option CELERYD_MAX_TASKS_PER_CHILD to avoid memory leaks.
+See: https://pulp.plan.io/issues/2172
+"""
+_PULP_MAX_TASKS_PER_CHILD = os.environ.get("PULP_MAX_TASKS_PER_CHILD")
+
+if _PULP_MAX_TASKS_PER_CHILD.isdigit() and _PULP_MAX_TASKS_PER_CHILD >= 1:
+    celery.conf.update(CELERYD_MAX_TASKS_PER_CHILD=_PULP_MAX_TASKS_PER_CHILD)
```

Comment:
At this point, even though `_PULP_TASKS_PER_CHILD` is `isidigit`, it's still a string. So...this happens:

```python
>>> '1' > 1
True
```

So `_PULP_MAX_TASKS_PER_CHILD` is always `>=` 1.

This should maybe be converted from LBYL to EAFP, e.g.
```python
try:
    _PULP_MAX_TASKS_PER_CHILD = int(os.environ.get("PULP_MAX_TASKS_PER_CHILD", 1)  # if not 1 here, then whatever a sane default is
    celery.conf.update(CELERYD_MAX_TASKS_PER_CHILD=_PULP_MAX_TASKS_PER_CHILD
except (TypeError, ValueError):
    # do something reasonable...maybe explode?
    # it seems like silently failing if the value in the config makes no sense is a bad idea, so
    # probably raising a good exception explaining why the value for this envvar is wrong would be good?
    do_something()
```